lib/Doctrine/ORM/EntityManager.php

```diff
@@ -157,8 +156,12 @@ protected function __construct(Connection $conn, Configuration $config, EventMan
         $this->metadataFactory->setCacheDriver($this->config->getMetadataCacheImpl());
 
         $this->repositoryFactory = $config->getRepositoryFactory();
-        $this->unitOfWork        = new UnitOfWork($this);
-        $this->proxyFactory      = new ProxyFactory(
+
+        $unitOfWorkClassName  = $config->getUnitOfWorkClassName();
+        $this->unitOfWork     = new $unitOfWorkClassName($this);
```

Comment:
@mcspronko what are your use case for customizing the UnitOfWork ? This is what you need to explain. We won't make it configurable without knowing why (and this is also why it should be in a dedicated PR).
And what is sensible in the UoW is both performance (it has a huge impact on the performance of Doctrine) and reliability (you should not start replacing the UoW implementation before you understand all the logic in the existing one. This will take time. We have lots of sensible logic in it)